SalesforceCommerceCloud/pwa-kit
# Feat: Implement `useStorageHelper` hook (@W-16300903@)

# Description

This PR adds the `useStorageHelper` hook that exposes helper functions that allow users to interact with data set by the `commerce-sdk-react` in local/cookie/memory storage. Currently, this hook exposes 1 storage helper, `clearStorage()`, which clears all data set in local/cookie storage by the `Auth` class.

This is useful for operations that require clearing/reseting storefront state.



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

1. `git checkout ju/expose-auth-helpers`
2. Run `npm run build` in `packages/commerce-sdk-react`
3. In the retail react app, pick a page/component of your choosing (for example homepage) and add the following lines:

```jsx
// imports
import {useProductSearch, useStorageHelper} from '@salesforce/commerce-sdk-react'

// Within the component
const {clearStorage} = useStorageHelper()

// Within the component return
return(
    ...
    <Button onClick={() => clearStorage()}>Clear Storage</Button>
)


```

4. Start the retail react app with `npm run start`
5. Navigate to the Clear storage button
6. Open up dev tools
7. click clear storage button and observe values set by the `Auth` class within cookies and local storage is cleared (You may have to click the refresh in the dev tools)

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
